eclass/linux-mod-r1.eclass

```diff
@@ -421,8 +423,16 @@ linux-mod-r1_src_compile() {
 	local -a emakeargs=( "${MODULES_MAKEARGS[@]}" )
 	[[ ${modargs@a} == *a* ]] && emakeargs+=( "${modargs[@]}" )
 
+	if use dkms; then
```

Comment:
Note that using linux-mod-r1_src_compile and src_install is optional (see eclassdoc's examples) and cannot be relied on, e.g. zfs-kmod does not use it for one and all that logic will be unused. It does require that packages call `modules_post_process` in src_install though.

The eclass also allows linux-mod-r1_src_compile to be called multiple times (noted in its description) which I guess would overwrite this dkms.conf.

Reply:
Ah, nevermind I was re-reading your post and you already had that listed in drawbacks for the first bit.